### ISSUE TITLE
Va-memorable-date property issues

### ISSUE THREAD

**Issue by nickjg231** [department-of-veterans-affairs/va.gov-team#61627]
## Issue Description
A support request came in for a user who is having issues with the `invalid-day`, `invalid-month`, and `invalid-year` properties in the `va-memorable-date` web component. He has linked a demo where the issue can be seen here:
https://codesandbox.io/s/bold-browser-xyct72?file=/src/App.js

The issue is that if `invalid-year` is set to true, he believes it is the only field that should show the error highlighting. This would apply to the other fields in the same way. Currently, when `invalid-year` is set to true, it toggles the aria-invalid property on the text input. The default error behavior currently will highlight all three inputs when there is an error present, and we don't distinguish between which inputs should highlight. This is something that we should discuss to see if it's an issue.

Original support request:
https://dsva.slack.com/archives/CBU0KDSB1/p1688756886472779

---
## Tasks
- [ ] Fix the issue experienced in `va-memorable-date`

## Acceptance Criteria
- [ ] The properties within `va-memorable-date` function properly
---
## How to configure this issue
- [ ] **Attached to a Milestone** (when will this be completed?)
- [ ] **Attached to an Epic** (what body of work is this a part of?)
- [ ] **Labeled with Team** (`product support`, `analytics-insights`, `operations`, `service-design`, `Console-Services`, `tools-fe`)
- [ ] **Labeled with Practice Area** (`backend`, `frontend`, `devops`, `design`, `research`, `product`, `ia`, `qa`, `analytics`, `contact center`, `research`, `accessibility`, `content`)
- [ ] **Labeled with Type** (`bug`, `request`, `discovery`, `documentation`, etc.)

**Comment by caw310** [comment]
@rsmithadhoc , please review before we work on this
